src/core/request/engines/provider/index.ts

```diff
@@ -0,0 +1,185 @@
+/*!
+ * V4Fire Core
+ * https://github.com/V4Fire/Core
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Core/blob/master/LICENSE
+ */
+
+import {
+
+	RequestEngine,
+	RequestOptions,
+	RequestResponse,
+
+	Response,
+	ResponseTypeValue
+
+} from 'core/request';
+import type iProvider from 'core/data/interface';
+import type { ProviderConstructor, ExtraProviderConstructor } from 'core/data/interface/types';
+import Provider, { providers, queryMethods } from 'core/data';
+import Then from 'core/then';
+import { concatUrls } from 'core/url';
+
+import type { AvailableOptions, MethodsMapping } from 'core/request/engines/provider/interface';
+import { availableParams } from 'core/request/engines/provider/const';
+
+export * from 'core/request/engines/provider/interface';
+export * from 'core/request/engines/provider/const';
+
+/**
+ * Returns provider class or object.
```

Comment:
Returns an instance of a data provider by the specified global name or constructor